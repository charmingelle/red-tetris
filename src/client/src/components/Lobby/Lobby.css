.lobby {
  width: 100vw;
  height: 100vh;
  max-height: 939px;
  display: flex;
}

.lobby-left {
  width: 70%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #666666;
}

.lobby-right {
  width: 30%;
  height: 100vh;
  background-color: #333;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.enter-name-form-container {
  margin-bottom: 50px;
}

.name-label {
  font-size: 36px;
  color: #f9f9f9;
}

.room-list {
  width: 700px;
  height: 90%;
  outline: 5px solid #f9f9f9;
  background-color: black;
  overflow-y: auto;
}

.no-rooms-label {
  margin: 20px;
  line-height: 55px;
  font-size: 36px;
  color: #f9f9f9;
}

.room-item {
  margin: 20px;
  display: flex;
  align-items: center;
}

.room-name {
  line-height: 55px;
  width: 440px;
  line-height: 55px;
  font-size: 36px;
  color: #f9f9f9;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  margin-right: 30px;
}

.room-actions {
  width: 170px;
  display: flex;
}

.game-in-progress-label {
  font-size: 30px;
  color: #129c0a;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.join-room-button {
  width: 100%;
  background-color: #c00;
  border: 5px solid;
  border-left-color: #f00;
  border-top-color: #f00;
  border-right-color: #980000;
  border-bottom-color: #980000;
  color: #fff;
  font-size: 30px;
  font-weight: bold;
  padding: 0 10px;
}

.join-room-button:focus {
  outline: none;
}
