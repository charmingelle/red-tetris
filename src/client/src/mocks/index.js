export const RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

export const LINE = [
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
];

export const GI = [
  [0, 0, 0],
  [1, 1, 1],
  [0, 0, 1],
];

export const EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];

export const NOT_EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const GAME_OVER_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, RED, RED, RED, RED, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const PILE_WITH_ROWS_TO_REMOVE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const TETRO = {
  figure: [GI],
  row: 5,
  col: 4,
  color: RED,
};

export const ROOM = {
  id: 'Room',
  leader: 'id-0',
  players: {
    'id-0': {
      id: 'id-0',
      name: 'Anna',
      pile: EMPTY_PILE,
      tetro: null,
      score: 0,
      isGameOver: false,
    },
  },
};

export const EMPTY_STORE = {
  socket: null,
  myId: null,
  myName: null,
  myRoom: null,
  people: {},
  rooms: {},
  tetro: null,
};

export const PERSON_ANNA = { name: 'Anna', score: 0 };

export const PERSON_OLYA = { name: 'Olya', score: 0 };
