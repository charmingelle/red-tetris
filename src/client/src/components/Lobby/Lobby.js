import React from 'react';
import { connect } from 'react-redux';
import './Lobby.css';
import { EnterNameForm } from '../EnterNameForm';
import { CreateRoomForm } from '../CreateRoomForm';
import { LobbyPeople } from '../LobbyPeople';

const joinRoom = (socket, id, myName) => () => {
  window.location.hash = `#${id}[${myName}]`;
  socket.emit('join-room', { roomId: id });
};

const RoomList = ({ myName, rooms, socket }) =>
  Object.values(rooms)
    .reverse()
    .map(({ id, game }) => (
      <div key={id} className="room-item">
        <div className="room-name">{id}</div>
        <div className="room-actions">
          {game ? (
            <div className="game-in-progress-label">Game on...</div>
          ) : (
            <button
              className="join-room-button"
              key={id}
              onClick={joinRoom(socket, id, myName)}
            >
              Join
            </button>
          )}
        </div>
      </div>
    ));

const RoomListBoard = ({ myName, rooms, socket }) => (
  <div className="room-list">
    {myName && Object.values(rooms).length ? (
      <RoomList myName={myName} rooms={rooms} socket={socket} />
    ) : myName ? (
      <div className="no-rooms-label">No rooms yet</div>
    ) : (
      <div className="no-rooms-label">No rooms available yet</div>
    )}
  </div>
);

export const LobbyInner = ({ myName, people, rooms, socket }) => (
  <div className="lobby">
    <div className="lobby-left">
      <RoomListBoard myName={myName} rooms={rooms} socket={socket} />
    </div>
    <div className="lobby-right">
      <div className="enter-name-form-container">
        {myName ? (
          <div className="name-label">{`Hi, ${myName}`}</div>
        ) : (
          <EnterNameForm />
        )}
      </div>
      {myName ? <CreateRoomForm /> : null}
      {myName && Object.values(people).length ? <LobbyPeople /> : null}
    </div>
  </div>
);

const mapStateToProps = ({ myName, people, rooms, socket }) => ({
  myName,
  people,
  rooms,
  socket,
});

export const Lobby = connect(mapStateToProps)(LobbyInner);
