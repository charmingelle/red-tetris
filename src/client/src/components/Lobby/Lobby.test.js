import React from 'react';
import { Lobby } from './Lobby';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { EMPTY_STORE, PERSON_ANNA, PERSON_OLYA } from '../../mocks';

const mockStore = configureStore([]);

test('Empty lobby', () => {
  const store = mockStore(EMPTY_STORE);

  expect(
    renderer
      .create(
        <Provider store={store}>
          <Lobby />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

test('Lobby with one player', () => {
  const store = mockStore({
    socket: null,
    myId: null,
    myName: 'Anna',
    myRoom: null,
    people: { 'id-0': PERSON_ANNA },
    rooms: {},
    tetro: null,
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <Lobby />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

test('Lobby with two players', () => {
  const store = mockStore({
    socket: null,
    myId: null,
    myName: 'Anna',
    myRoom: null,
    people: {
      'id-0': PERSON_ANNA,
      'id-1': PERSON_OLYA,
    },
    rooms: {},
    tetro: null,
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <Lobby />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

test('Lobby with two players and one roon ready for joining', () => {
  const store = mockStore({
    socket: null,
    myId: null,
    myName: 'Anna',
    myRoom: null,
    people: {
      'id-0': PERSON_ANNA,
      'id-1': PERSON_OLYA,
    },
    rooms: {
      Room: { id: 'Room', game: null },
    },
    tetro: null,
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <Lobby />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

test('Lobby with two players and one roon with game on', () => {
  const store = mockStore({
    socket: null,
    myId: null,
    myName: 'Anna',
    myRoom: null,
    people: {
      'id-0': PERSON_ANNA,
      'id-1': PERSON_OLYA,
    },
    rooms: {
      Room: { id: 'Room', game: {} },
    },
    tetro: null,
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <Lobby />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
