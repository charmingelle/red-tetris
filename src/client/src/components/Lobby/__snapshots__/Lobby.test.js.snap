// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty lobby 1`] = `
<div
  className="lobby"
>
  <div
    className="lobby-left"
  >
    <div
      className="room-list"
    >
      <div
        className="no-rooms-label"
      >
        No rooms available yet
      </div>
    </div>
  </div>
  <div
    className="lobby-right"
  >
    <div
      className="enter-name-form-container"
    >
      <div
        className="enter-name-form"
      >
        <div
          className="enter-name-label"
        >
          What's your name?
        </div>
        <input
          className="enter-name-input"
          maxLength="15"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Enter name..."
          type="text"
          value=""
        />
        <button
          className="enter-name-button"
          onClick={[Function]}
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lobby with one player 1`] = `
<div
  className="lobby"
>
  <div
    className="lobby-left"
  >
    <div
      className="room-list"
    >
      <div
        className="no-rooms-label"
      >
        No rooms yet
      </div>
    </div>
  </div>
  <div
    className="lobby-right"
  >
    <div
      className="enter-name-form-container"
    >
      <div
        className="name-label"
      >
        Hi, Anna
      </div>
    </div>
    <div
      className="create-room-form"
    >
      <div
        className="create-room-label"
      >
        Create own room
      </div>
      <input
        className="create-room-input"
        maxLength="15"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter name..."
        type="text"
        value=""
      />
      <button
        className="create-room-button"
        onClick={[Function]}
      >
        Create
      </button>
    </div>
    <div
      className="lobby-people"
    >
      <div
        className="lobby-people-label"
      >
        Players
      </div>
      <div
        className="lobby-people-list"
      >
        <div
          className="lobby-person-name"
          style={
            Object {
              "color": "#f9f9f9",
            }
          }
        >
          Anna - 0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lobby with two players 1`] = `
<div
  className="lobby"
>
  <div
    className="lobby-left"
  >
    <div
      className="room-list"
    >
      <div
        className="no-rooms-label"
      >
        No rooms yet
      </div>
    </div>
  </div>
  <div
    className="lobby-right"
  >
    <div
      className="enter-name-form-container"
    >
      <div
        className="name-label"
      >
        Hi, Anna
      </div>
    </div>
    <div
      className="create-room-form"
    >
      <div
        className="create-room-label"
      >
        Create own room
      </div>
      <input
        className="create-room-input"
        maxLength="15"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter name..."
        type="text"
        value=""
      />
      <button
        className="create-room-button"
        onClick={[Function]}
      >
        Create
      </button>
    </div>
    <div
      className="lobby-people"
    >
      <div
        className="lobby-people-label"
      >
        Players
      </div>
      <div
        className="lobby-people-list"
      >
        <div
          className="lobby-person-name"
          style={
            Object {
              "color": "#f9f9f9",
            }
          }
        >
          Anna - 0
        </div>
        <div
          className="lobby-person-name"
          style={
            Object {
              "color": "#f9f9f9",
            }
          }
        >
          Olya - 0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lobby with two players and one roon ready for joining 1`] = `
<div
  className="lobby"
>
  <div
    className="lobby-left"
  >
    <div
      className="room-list"
    >
      <div
        className="room-item"
      >
        <div
          className="room-name"
        >
          Room
        </div>
        <div
          className="room-actions"
        >
          <button
            className="join-room-button"
            onClick={[Function]}
          >
            Join
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="lobby-right"
  >
    <div
      className="enter-name-form-container"
    >
      <div
        className="name-label"
      >
        Hi, Anna
      </div>
    </div>
    <div
      className="create-room-form"
    >
      <div
        className="create-room-label"
      >
        Create own room
      </div>
      <input
        className="create-room-input"
        maxLength="15"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter name..."
        type="text"
        value=""
      />
      <button
        className="create-room-button"
        onClick={[Function]}
      >
        Create
      </button>
    </div>
    <div
      className="lobby-people"
    >
      <div
        className="lobby-people-label"
      >
        Players
      </div>
      <div
        className="lobby-people-list"
      >
        <div
          className="lobby-person-name"
          style={
            Object {
              "color": "#f9f9f9",
            }
          }
        >
          Anna - 0
        </div>
        <div
          className="lobby-person-name"
          style={
            Object {
              "color": "#f9f9f9",
            }
          }
        >
          Olya - 0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lobby with two players and one roon with game on 1`] = `
<div
  className="lobby"
>
  <div
    className="lobby-left"
  >
    <div
      className="room-list"
    >
      <div
        className="room-item"
      >
        <div
          className="room-name"
        >
          Room
        </div>
        <div
          className="room-actions"
        >
          <div
            className="game-in-progress-label"
          >
            Game on...
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="lobby-right"
  >
    <div
      className="enter-name-form-container"
    >
      <div
        className="name-label"
      >
        Hi, Anna
      </div>
    </div>
    <div
      className="create-room-form"
    >
      <div
        className="create-room-label"
      >
        Create own room
      </div>
      <input
        className="create-room-input"
        maxLength="15"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter name..."
        type="text"
        value=""
      />
      <button
        className="create-room-button"
        onClick={[Function]}
      >
        Create
      </button>
    </div>
    <div
      className="lobby-people"
    >
      <div
        className="lobby-people-label"
      >
        Players
      </div>
      <div
        className="lobby-people-list"
      >
        <div
          className="lobby-person-name"
          style={
            Object {
              "color": "#f9f9f9",
            }
          }
        >
          Anna - 0
        </div>
        <div
          className="lobby-person-name"
          style={
            Object {
              "color": "#f9f9f9",
            }
          }
        >
          Olya - 0
        </div>
      </div>
    </div>
  </div>
</div>
`;
